application/src/main/java/org/togetherjava/tjbot/features/filesharing/FileSharingMessageListener.java

```diff
@@ -104,13 +89,41 @@ public void onMessageReceived(MessageReceivedEvent event) {
             try {
                 processAttachments(event, attachments);
             } catch (Exception e) {
-                LOGGER.error(
+                logger.error(
                         "Unknown error while processing attachments. Channel: {}, Author: {}, Message ID: {}.",
                         event.getChannel().getName(), author.getId(), event.getMessageId(), e);
             }
         });
     }
 
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
+        Member interactionUser = event.getMember();
+        String gistAuthorId = args.get(0);
+        boolean hasSoftModPermissions =
+                interactionUser.getRoles().stream().map(Role::getName).anyMatch(isSoftModRole);
+
+        if (!gistAuthorId.equals(interactionUser.getId()) && !hasSoftModPermissions) {
+            event.reply("You do not have permission for this action.").setEphemeral(true).queue();
+            return;
+        }
+
+        Message message = event.getMessage();
+        List<Button> buttons = message.getButtons();
+        event.editComponents(ActionRow.of(buttons.stream().map(Button::asDisabled).toList()))
+            .queue();
+
+        String gistId = args.get(1);
+
+        try {
+            new GitHubBuilder().withOAuthToken(gistApiKey).build().getGist(gistId).delete();
+
+            event.getMessage().delete().queue();
+        } catch (IOException e) {
+            logger.error("Failed to delete gist with id {}", gistId);
```

Comment:
use `warn`, its not an error. why IOException? the code previously had the log only if the file was 404, how does this api-wrapper expose that information?

Reply:
it throws an io exception on github object creation already

Reply:
maybe if the token is wrong

Reply:
mh okay. warn though please. thanks

Reply:
done